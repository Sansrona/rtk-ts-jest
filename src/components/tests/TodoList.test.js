import * as reduxHooks from '../../hooks'
import { render } from '@testing-library/react';

import TodoList from "../TodoList";

jest.mock('../../hooks');

const todos = [
    {id: '123', title: 'React', completed: false},
    {id: '124', title: 'Redux', completed: false},
]

describe('TodoList', () => {
    it('should render a TodoList', () => {
        jest.spyOn(reduxHooks, 'useAppSelector').mockReturnValue([]); 
        const { view } = render(<TodoList />);


        expect(view).toMatchSnapshot();
    });
    it('should render a TodoList with items', () => {
        jest.spyOn(reduxHooks, 'useAppSelector').mockReturnValue(todos); 
        const { view } = render(<TodoList />);


        expect(view).toMatchSnapshot();
    })
})
