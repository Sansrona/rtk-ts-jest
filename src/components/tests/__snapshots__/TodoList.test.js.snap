// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render a TodoList 1`] = `undefined`;

exports[`TodoList should render a TodoList with items 1`] = `undefined`;
